### PR TITLE
fix(tmc): outputs sharing failure cases not synced to TMC.

### PR DESCRIPTION
**Pull request by i4ki**
## What this PR does / why we need it:

It fixes the `outputs-sharing` experiment not synchronizing failure cases to Terramate Cloud.

## Which issue(s) this PR fixes:
none
## Does this PR introduce a user-facing change?
```
yes, fixes a bug in v0.10.1
```

### CODE REVIEW

**Review by snakster — COMMENTED** [review]
I think there are still problems in this function. Some of them were already there before, so this was my mistake, but it has been copied.

I think it's never ok to return from the inner function:
```
c.cloudSyncAfter(...)
...
return errs.AsError()
```
The problem is that `c.cloudSyncAfter` will only sync, if the task is the cloud sync command. For scripts, this means if an earlier command fails which is not the cloud sync command (i.e. terraform init), even though it goes through the `cloudSyncAfter(...)`, it will not sync.

Cancel does it correctly:
```
select {
case <-cancelCtx.Done():
	c.cloudSyncAfter(cloudRun, runResult{ExitCode: -1}, errors.E(ErrRunCanceled))
	continue
default:
}
```

**Review by soerenmartius — APPROVED** [review]
LGTM

File: cmd/terramate/cli/run.go [hunk 1]
```diff
@@ -385,10 +386,23 @@ func (c *cli) runAll(
...
+			// For cloud sync, we always assume that there's a single task per stack.
+			cloudRun := stackCloudRun{Stack: run.Stack, Task: task}
+
+			select {
+			case <-cancelCtx.Done():
+				c.cloudSyncAfter(cloudRun, runResult{ExitCode: -1}, errors.E(ErrRunCanceled))
+				releaseResource()
+				continue tasksLoop
+			default:
+			}
```
Comment (i4ki) [inline cmd/terramate/cli/run.go:404]: This is moved up so we don't try any outputs-sharing handling if the stack execution is already canceled.

**Review by snakster — APPROVED** [review]
Looks good now 👍

**Pull request merged**